src/java.base/share/classes/java/lang/reflect/Executable.java

```diff
@@ -337,17 +340,24 @@ Type[] getAllGenericParameterTypes() {
             if (realParamData) {
                 final Type[] out = new Type[nonGenericParamTypes.length];
                 final Parameter[] params = getParameters();
-                int fromidx = 0;
-                for (int i = 0; i < out.length; i++) {
+                int fromidx = genericParamTypes.length - 1;
```

Comment:
so provided that synthetic parameters are first, then mandated parameters if any and finally explicit parameters, I think that it is safe to get as many parameters as possible from the genericParamTypes array and if this array is shorter than the nonGenericParamTypes array, then the difference must be due to the presence of synthetic parameters, and we take the type of those from the nonGenericParamTypes array

Reply:
Do synthetic parameters really come before mandated ones? Seems not the case for anonymous classes in instance methods that capture local variables from its enclosing method, those captured local variables are passed as synthetic parameters in the end of the pareamter list while the mandated enclosing instance is the first parameter.